### PR TITLE
docs: add returning on the first error example for try_join!

### PR DESCRIPTION
**Pull request by hi-rustin**
Signed-off-by: hi-rustin <rustin.liu@gmail.com>

## Motivation

close https://github.com/tokio-rs/tokio/issues/4127
## Solution

 add returning on the first error example for try_join!

### CODE REVIEW

File: tokio/src/macros/try_join.rs [hunk 1]
```diff
@@ -59,6 +59,44 @@
 ///     }
 /// }
 /// ```
+///
+/// Returning on the first error.
```
Comment (Darksonn) [inline tokio/src/macros/try_join.rs:63]: I think this is a poor description of the example.
```suggestion
/// Using `try_join!` with spawned tasks.
```

Comment (hi-rustin) [reply tokio/src/macros/try_join.rs]: Thanks for your review!  💚 💙 💜 💛 ❤️ Addressed!

File: tokio/src/macros/try_join.rs [hunk 2]
```diff
@@ -59,6 +59,44 @@
...
+/// #[tokio::main]
+/// async fn main() {
+///     let handle1 = tokio::spawn(do_stuff_async());
+///     let handle2 = tokio::spawn(more_async_work());
+///     match tokio::try_join!(flatten(handle1), flatten(handle2)) {
+///         Ok(val) => {
+///             // do something with the values
+///         }
+///         Err(val) => {
+///             assert_eq!(val, "failed");
```
Comment (Darksonn) [inline tokio/src/macros/try_join.rs:95]: I think the example would be better like this.
```suggestion
///         Err(err) => {
///             println!("Failed with {}.", err);
///             # assert_eq!(val, "failed");
```

**Review by Darksonn — APPROVED** [review]

**Comment by Darksonn** [comment]
Thanks!

**Pull request merged**